app/src/main/java/org/cru/godtools/ui/dashboard/tools/GridConcatAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package org.cru.godtools.ui.dashboard.tools
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import org.ccci.gto.android.common.recyclerview.adapter.SimpleDataBindingAdapter
+import org.cru.godtools.databinding.GridConcatViewBinding
+
+class GridConcatAdapter(
+    lifecycleOwner: LifecycleOwner,
+    private val recyclerAdapter: RecyclerView.Adapter<*>,
+    private val rowCount: Int = 1,
+    private val titleText: String,
+    private val subTitleText: String? = null,
+    private val showDivider: Boolean = false
+) : SimpleDataBindingAdapter<GridConcatViewBinding>(lifecycleOwner) {
```

Comment:
I think you are trying to reuse too much here and making the layout much more dependent on code than it probably should be. 

you should have 2 distinct layouts, 1 for the spotlight section and 1 for the category selectors. All the configuration for those layouts should be able to be done purely in XML and you shouldn't need to pass in `rowCount`, `titleText`, `subTitleText`, or `showDivider`.

both of these layouts would have a RecyclerView that accepts an adapter, the spotlight one would be configured to use a horizontal `LinearLayoutManager` in the xml, and the categories one would use `GridLayoutManager` with 2 rows configured in the XML.